src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +83,13 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex() == Person.Sex.WOMAN && p.getAge() >= femaleAge
+                        && p.getCats().size() > 0)
+                .map(Person::getCats)
+                .flatMap(List::stream)
```

Comment:
```suggestion
                .map(Person::getCats)
                .flatMap(List::stream)
```

combine into one flatMap